terraform/data.tf

```diff
@@ -291,3 +291,96 @@ data "aws_iam_policy_document" "s3_alb_policy" {
     }
   }
 }
+
+# S3 snapshot bucket
+data "aws_iam_policy_document" "s3bucket_policy" {
+  count  = terraform.workspace == "stage" ? 1 : 0
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:ListBucket",
+        "s3:GetObject",
+        "s3:PutObject",
+        "s3:DeleteObject",
+        "s3:ListBucketVersions",
+        "s3:GetObjectVersion"
+      ],
+      "Resource": [
+        "arn:aws:s3:::${local.s3_snapshot_bucket_name}",
+        "arn:aws:s3:::${local.s3_snapshot_bucket_name}/*"
+      ]
+    }
+  ]
+}
+
+#Opensearch snapshot policy
+
+data "aws_iam_policy_document" "trust" {
+  statement {
```

Comment:
This is supposed to only be created for the OpenSearch cluster that is creating the snapshot, which is the dev cluster. So use the following:

count = terraform.workspace == "dev" ? 1 : 0